optimum/intel/openvino/modeling.py

```diff
@@ -1030,3 +1040,356 @@ def forward(self, **kwargs):
+
+    def to(self, device: str):
+        """
+        Use the specified `device` for inference. For example: "cpu" or "gpu". `device` can
+        be in upper or lower case. To speed up first inference, call `.compile()` after `.to()`.
+        """
+        self.text_model.to(device=device)
+        self.visual_model.to(device=device)
+
+        return self
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        export: bool = False,
+        config: Optional["PretrainedConfig"] = None,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: str = HUGGINGFACE_HUB_CACHE,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        task: Optional[str] = None,
+        trust_remote_code: bool = False,
+        use_separate_models=True,
+        **kwargs,
+    ):
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        text_model = OVModelCLIPText.from_pretrained(
+            model_id=model_id,
+            export=export,
+            config=config,
+            token=token,
+            revision=revision,
+            force_download=force_download,
+            cache_dir=cache_dir,
+            subfolder=subfolder,
+            local_files_only=local_files_only,
+            task=task,
+            trust_remote_code=trust_remote_code,
+        )
+
+        visual_model = OVModelCLIPVisual.from_pretrained(
+            model_id=model_id,
+            export=export,
+            config=config,
+            token=token,
+            revision=revision,
+            force_download=force_download,
+            cache_dir=cache_dir,
+            subfolder=subfolder,
+            local_files_only=local_files_only,
+            task=task,
+            trust_remote_code=trust_remote_code,
+        )
+
+        return cls(text_model=text_model, visual_model=visual_model, config=config)
+
+    def forward(
+        self,
+        input_ids: Union[torch.Tensor, np.ndarray],
+        pixel_values: Union[torch.Tensor, np.ndarray],
+        **kwargs,
+    ):
+        self.text_model.compile()
+        self.visual_model.compile()
+
+        text_model_outputs = self.text_model._inference({"text": input_ids})
+        visual_model_outputs = self.visual_model._inference({"x": pixel_values})
+
+        return ModelOutput(
+            image_features=visual_model_outputs["image_features"], text_features=text_model_outputs["text_features"]
```

Comment:
I believe zero shot classification means that you need return not only features, but also probabilities like here https://github.com/huggingface/transformers/blob/main/src/transformers/models/clip/modeling_clip.py#L1334

BTW, can these model classes can also handle transformers clip realization? if only open_clip expected, then it is better rename model class

Reply:
names are updated, output are updated too according to transformers clip model output